### PR TITLE
Fix pow eigvals

### PR DESCRIPTION
**Pull request by cwgambla**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
TLDR: qml.eigvals returns nan for a legal fractional power operator

**Description of the Change:**
Casted complex over eigenvals in pow.py

**Benefits:**
Fixes the bug, qml.eigvals no longer returns nan for legal fractional power operator

**Possible Drawbacks:**
Has not been tested for affecting performance and parallelization compatibility. 

**Related GitHub Issues:**
https://github.com/PennyLaneAI/pennylane/issues/9632

### CODE REVIEW

**Comment by comp-phys-marc** [comment]
Hello! Thanks for your great contribution. That is a good catch!

If you could please merge `main`, it will help us to get past a known failure in the CI that is currently showing up due to a downstream dependency updating its version.

Then I am happy to approve this and get it in! I would just ask that you please add your name and an entry to `changelog-dev.md` please. :)

File: tests/ops/op_math/test_pow_op.py [hunk 1]
```diff
@@ -441,6 +441,17 @@ def test_adjoint_non_integer_power_raises(self, z, power_method):
...
+    def test_eigvals_fractional_power_negative_eigenvalue(self, z, power_method):
+        """Test pow method works for complex numbers"""
+        base = qp.PauliZ(0)
+        op = power_method(base=base, z=z) 
+        
+        eigvals = op.eigvals()
+        
+        """Expected eigenvalues for Z**0.5 are [1.0, 1.0j]"""
+        expected_eigvals = [1.0, 1j]
+        assert np.allclose(eigvals, expected_eigvals)
```
Comment (andrijapau) [inline tests/ops/op_math/test_pow_op.py]: If you do want to parametrize (and I think it's worth doing), perhaps we can do something like,

```
@pytest.mark.parametrize("z", [0.5, 1.5, -0.5])
def test_eigvals_fractional_power_negative_eigenvalue(self, z, power_method):
    """Test that the pow method correctly calculates complex eigenvalues 
    for various fractional powers of an operator."""

    base = qp.PauliZ(0)
    op = power_method(base=base, z=z) 
    
    eigvals = op.eigvals()
    
    expected_eigvals = np.array([1.0**z, (-1.0 + 0j)**z])
    
    assert np.allclose(eigvals, expected_eigvals)
```

**Review by andrijapau — COMMENTED** [review]
Looks good to me. 🚀 

Please add a changelog entry and yourself as a contributor and I'd be happy to approve (provided CI is happy)!

**Comment by comp-phys-marc** [comment]
Looks like we just need to fix some formatting and then things will be good to go :+1: !

**Comment by CatalinaAlbornoz** [comment]
@cwgambla let us know if you have any questions about the review comments!

**Review by sankalpsthakur — DISMISSED** [review]
`complex(value)` forces an interface value through Python scalar conversion, which regresses traced parameter-dependent eigenspectra. On this PR head (`5429fd1`), this minimal case:

```python
import jax
import jax.numpy as jnp
import pennylane as qp

@jax.jit
def powered_eigvals(x):
    return jnp.stack(qp.pow(qp.s_prod(x, qp.X(0)), 2).eigvals())

print(powered_eigvals(0.5))
```

raises `jax.errors.ConcretizationTypeError` at `pow.py:331`: "The problem arose with the `complex` function." The same script returns `[0.25 0.25]` on current `main`, so this affects integer powers as well as the fractional-negative case being fixed.

Please use interface-aware complex promotion (for example via `qp.math` rather than Python `complex`) and add a JAX-jit regression using a parameterized base such as `SProd`.

**Comment by sankalpsthakur** [comment]
Tool-assistance disclosure for my review above: OpenAI Codex assisted in preparing the reproduction and wording. I reran and verified both the PR-head failure and the current-main result before submitting the review.

**Comment by andrijapau** [comment]
> `complex(value)` forces an interface value through Python scalar conversion, which regresses traced parameter-dependent eigenspectra. On this PR head (`5429fd1`), this minimal case:
> 
> ```python
> import jax
> import jax.numpy as jnp
> import pennylane as qp
> 
> @jax.jit
> def powered_eigvals(x):
>     return jnp.stack(qp.pow(qp.s_prod(x, qp.X(0)), 2).eigvals())
> 
> print(powered_eigvals(0.5))
> ```
> 
> raises `jax.errors.ConcretizationTypeError` at `pow.py:331`: "The problem arose with the `complex` function." The same script returns `[0.25 0.25]` on current `main`, so this affects integer powers as well as the fractional-negative case being fixed.
> 
> Please use interface-aware complex promotion (for example via `qp.math` rather than Python `complex`) and add a JAX-jit regression using a parameterized base such as `SProd`.

Hi @sankalpsthakur , thank your for your input and great catch! Your intuition is correct, something like,
```python
qp.math.cast(value, dtype=np.complex128)
```
would work nicely instead, as it is interface-aware. 

@cwgambla , could we please make this change and add an appropriate regression test? 👍🏼 

Please let me know if anything above is unclear or if you have any questions.

**Comment by sankalpsthakur** [comment]
Thanks for updating this. I reran the original JAX-jit reproduction at `139f961`; it now returns `[0.25+0.j, 0.25-0.j]`, so the concretization failure is fixed. A few items still keep the original request open:

- The requested JAX-jit regression with a parameterized `SProd` base is still missing; the current test only covers a static `PauliZ` at `z=0.5`.
- With default JAX (`jax_enable_x64=False`), `cast(..., dtype="complex128")` emits two truncation warnings and produces `complex64`. A precision-aware promotion such as `((1 + 0j) * value) ** self.z` avoids requesting unsupported `complex128` and is consistent with existing repository patterns.
- `sprod.py` now contains an unrelated commented-out duplicate of `SProd.label`; please remove that block.

Once those are addressed, I’m happy to recheck.

**Comment by andrijapau** [comment]
> Thanks for updating this. I reran the original JAX-jit reproduction at `139f961`; it now returns `[0.25+0.j, 0.25-0.j]`, so the concretization failure is fixed. A few items still keep the original request open:
> 
>     * The requested JAX-jit regression with a parameterized `SProd` base is still missing; the current test only covers a static `PauliZ` at `z=0.5`.
> 
>     * With default JAX (`jax_enable_x64=False`), `cast(..., dtype="complex128")` emits two truncation warnings and produces `complex64`. A precision-aware promotion such as `((1 + 0j) * value) ** self.z` avoids requesting unsupported `complex128` and is consistent with existing repository patterns.
> 
>     * `sprod.py` now contains an unrelated commented-out duplicate of `SProd.label`; please remove that block.
> 
> 
> Once those are addressed, I’m happy to recheck.

Hello @sankalpsthakur, and thanks again for catching the JAX jit compatibility bug! 🚀 

We deeply appreciate your time and effort but to keep the review moving efficiently, we're going to limit reviews to be from official codebase maintainers. 

Thanks for understanding and let us know if you have any questions. 🙏🏼

**Comment by sankalpsthakur** [comment]
Totally fair on maintainer reviews — no worries.

Quick confirm on `139f961`: JAX-jit concretization is fixed, but my change request still stands:

1. JAX-jit regression with parameterized `SProd` (not just static `PauliZ`)
2. Swap `cast(..., complex128)` for precision-aware promotion — default JAX truncates to complex64 with warnings
3. Drop the commented-out `SProd.label` block in `sprod.py`

Happy for maintainers to take it from here once those land.

**Comment by Alex-Preciado** [comment]
@sankalpsthakur, to reinforce what @andrijapau already mentioned, please do not leave change requests or review requirements on other contributors' PRs. To avoid confusing community members and keep our review queue manageable, maintainers will evaluate and triage PRs directly. Thank you for respecting this boundary moving forward.

**Comment by cwgambla** [comment]
Should I attempt to make the remaining fixes? Or leave them be for now

**Comment by comp-phys-marc** [comment]
> Should I attempt to make the remaining fixes? Or leave them be for now

If you have the capacity and availability to make them, that would be super! :)

**Comment by cwgambla** [comment]
I managed to change issues:
"Swap cast(..., complex128) for precision-aware promotion — default JAX truncates to complex64 with warnings"
"Drop the commented-out SProd.label block in sprod.py"

I am not so sure about:
"JAX-jit regression with parameterized SProd (not just static PauliZ)"

**Comment by comp-phys-marc** [comment]
Thanks so much @cwgambla ! Let me take a look...

File: tests/ops/op_math/test_pow_op.py [hunk 2]
```diff
@@ -435,7 +435,19 @@ def test_adjoint_non_integer_power_raises(self, z, power_method):
         with pytest.raises(AdjointUndefinedError, match="The adjoint of Pow operators"):
             _ = op.adjoint()
 
-
+    @pytest.mark.parametrize("z", [0.5, 1.5, -0.5])
```
Comment (andrijapau) [inline tests/ops/op_math/test_pow_op.py:438]: Nice! 🚀

File: tests/ops/op_math/test_pow_op.py
Comment (andrijapau) [inline tests/ops/op_math/test_pow_op.py:1]: Could we just add a regression test to ensure that the `eigvals` method is JAX-jittable?

Something simple like this should suffice,
```python
@pytest.mark.jax
def test_pow_eigvals_is_jittable(self):
    """Test that the eigvals method is jittable."""
    import jax
    import jax.numpy as jnp
    import numpy as np
    import pennylane as qp
    
    @jax.jit
    def f(x):
        return jnp.array(qp.pow(qp.RX(x, 0), 2).eigvals())

    x = 0.5
    expected = np.array([
        np.cos(x) + np.sin(x) * 1j, 
        np.cos(x) - np.sin(x) * 1j
    ])
    
    assert np.allclose(f(x), expected)
```

**Review by andrijapau — CHANGES_REQUESTED** [review]
Awesome! It's looking very close!

**Comment by andrijapau** [comment]
Hi @cwgambla , please let me know if you need any further clarifications on my review above. 🙏🏼

**Comment by CatalinaAlbornoz** [comment]
Hi @cwgambla , we haven't heard from you in a while. Are you still able to work on the last changes suggested by our team?

**Comment by CatalinaAlbornoz** [comment]
Hi @cwgambla, just checking in on this!

If you're still interested in continuing but just need some more time, just let us know. Otherwise, we'll close this PR in about a week due to inactivity. We can easily reopen it later whenever you're ready to pick it back up.

**Comment by comp-phys-marc** [comment]
@andrijapau I've added the test you requested and confirmed it passes. Any other thoughts before we merge?

**Comment by andrijapau** [comment]
> @andrijapau I've added the test you requested and confirmed it passes. Any other thoughts before we merge?

We should ensure `Pow2` has the change as well

**Comment by comp-phys-marc** [comment]
Alight @andrijapau, that was a very good point! I've added the fix to `Pow2` now and tested that as well. :)

**Comment by comp-phys-marc** [comment]
Let me just address these failures, it is because eigvals' signature has changed.

**Review by andrijapau — APPROVED** [review]
Just needs a changelog please!

**Comment by comp-phys-marc** [comment]
Hello @cwgambla we were able to find some capacity to work on this internally. Since you got this so close to a complete contribution, we will be able to get it over the line from this point! 🙂 You will be given attribution in the next release of PennyLane for your contribution! Thanks again.

File: tests/ops/op_math/test_pow2.py [hunk 3]
```diff
@@ -332,3 +339,42 @@ def test_pow_rotation2():
     assert pow_rotation.compute_resources(**op.arguments) == Resources(
         {OneWireDynOp(Float, wires=Wire[1]): 1}
     )
+
+
+@pytest.mark.parametrize(
+    "z, power_method", [(0.5, Pow2), (1.5, pow_using_dunder_method), (-0.5, qp.pow)]
```
Comment (astralcai) [inline tests/ops/op_math/test_pow2.py:345]:
```suggestion
    "z, power_method", [(0.5, Pow2), (1.5, lambda a, b: a ** b), (-0.5, qp.pow)]
```
This is optional but I don't think `base**z` is worth defining a dedicated helper function for.

**Review by astralcai — APPROVED** [review]
One non-blocking comment but overall looks good to me.
